source/agora/test/DNS.d

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+
+    Copyright:
+        Copyright (c) 2019-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.DNS;
+
+version (unittest):
+
+import agora.test.Base;
+import geod24.LocalRest;
+import geod24.concurrency;
+
+/// DNS test
+unittest
+{
+    import std.stdio;
+
+    TestConf conf;
+    auto network = makeTestNetwork!TestAPIManager(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    FiberScheduler scheduler = new FiberScheduler;
+    scheduler.start(
+        () {
+            auto addr = network.resolve(Address("agora://boa1xzval2a3cdxv28n6slr62wlczslk3juvk7cu05qt3z55ty2rlfqfc6egsh2.validators.unittest.bosagora.io"));
```

Comment:
Check the value that is returned ? Also we probably need a retry as the nodes might not have registered themselves even at the end of discovery (although that is unlikely).